### DISCUSSION TITLE
Not getting anything logged!

### DISCUSSION THREAD

**Discussion by nemesyssoft** [CocoaLumberjack/CocoaLumberjack#1364] [Q&A] [answered]
I'm starting to use CocoaLumberjack. I think I pretty much followed the instructions in the Getting Started:

In my pch file, I have:

#define LOG_LEVEL_DEF ddLogLevel
#import <CocoaLumberjack/CocoaLumberjack.h>
static const DDLogLevel ddLogLevel = DDLogLevelVerbose;

I'm using DDLogDebug directives in all my code but I don't see anything logged in the Xcode console. I must be doing something wrong?

I set a breakpoint in the viewDidLoad of my initial view controller and then checked the ddLogLevel, just to be sure it was set:

(lldb) p ddLogLevel
(const DDLogLevel) DDLogLevelVerbose

So, at this point, I'm stumped. Anybody has any suggestion on what I might be missing?

**Comment by ffried** [comment]
Which loggers have you added? And with which configuration?

**Comment by nemesyssoft** [comment]
I pretty much followed the instructions in Getting Started, so in `applicationDidFinishLaunching:`, I have:
`[DDLog addLogger:[DDOSLogger sharedInstance]];`
I know I'm missing something but I don't know what it is. I've been using CocoaLumberjack successfully in another project but that was for iOS in Swift. This one is in Obj-C for macOS.

**Reply by ffried** [reply]
Can you try to reproduce it in another project? Maybe add a small custom logger and see if your log messages are getting there?

**Reply by ffried** [reply]
Also, maybe try a `DDLogVerbose` or `DDLogWarn` just to try out if that works.

**Reply by ffried** [reply]
Last but not least you can try to use `DDLogLevelAll` as `ddLogLevel`.

**Reply by nemesyssoft** [reply, marked as answer]
All right, so, in applicationDidFinishLaunching:, just after the `[DDLog addLogger:[DDOSLogger sharedInstance]];`, I added:
`DDLogVerbose(@"Starting…");`

And I now see it in the Xcode console:
```
Starting…
Timestamp: 2023-08-28 12:03:42.070199-04:00
```
 I'll try to change the log level to DDLogLevelAll...

**Reply by ffried** [reply]
That sounds like it's already working then. `DDLogLevelAll` is not strictly necessary. `DDLogLevelVerbose` should be enough (I just double checked and your test shows it as well. The issue with your other log statements must lie elsewhere then.

**Reply by nemesyssoft** [reply]
Something a little odd I just realized. I used a bunch of logging using NSLog(). Even though my now `DDLogDebug(@"Starting…");` appears in the Xcode console, it shows up *after* the NSLog directives, even though it is issued in **applicationDidFinishLaunching:**.

How is that possible:
![Screenshot 2023-08-28 at 12 09 07](https://github.com/CocoaLumberjack/CocoaLumberjack/assets/2309889/67a0cbf7-8660-47f8-912b-28a52ace9f65)

All the other directives come way after the application started.

**Reply by ffried** [reply]
This is likely due to the fact that CocoaLumberjack has internal queues to which log messages are eventually dispatched asynchronously, while NSLog likely is synchronous.

**Reply by nemesyssoft** [reply]
Ah ok, thanks!
